showcase/app/components/mock/app/main/generic-advanced-table.gts

```diff
@@ -0,0 +1,599 @@
+    created: 'Feb 15 2025',
+    updated: 'Feb 15 2025',
+  },
+  {
+    name: 'fu-ip-ta',
+    'project-name': 'ca reba be',
+    'current-run-id': 'run-19Yks9WCFeD9xRTWo',
+    'run-status': 'applied',
+    'run-status-color': HdsBadgeColorValues.Success,
+    'current-run-applied': 'Mar 06, 2025 08:57:14 am',
+    'vcs-repo': 'example/gt]5*c!N1*N%I!m)18',
+    'module-count': 158,
+    modules: 'wad-bedzeaje-rogmejca',
+    'provider-count': 11,
+    providers: 'susnup-da-zuw',
+    'terraform-version': '0.14.0',
+    'state-terraform-version': '0.15.0',
+    created: 'Feb 14 2025',
+    updated: 'Feb 14 2025',
+  },
+];
+
+const updateModelWithSelectAllState = (
+  modelData: HdsAdvancedTableSignature['Args']['model'],
+  selectAllState: boolean
+) => {
+  modelData.forEach((modelRow) => {
+    modelRow['isSelected'] = selectAllState;
+  });
+};
+
+const updateModelWithSelectableRowsStates = (
+  modelData: HdsAdvancedTableSignature['Args']['model'],
+  selectableRowsStates: HdsAdvancedTableOnSelectionChangeSignature['selectableRowsStates']
+) => {
+  const modelDataMap = new Map(
+    modelData.map((modelRow) => [modelRow['id'], modelRow])
+  );
+  selectableRowsStates.forEach((row) => {
+    // safe to assume that there is always a record for the "selectionKey" since it's coming from the model (the selectable "rows" are a subset of the model dataset)
+    const rowFromModel = modelDataMap.get(row.selectionKey);
+    if (rowFromModel) {
+      rowFromModel['isSelected'] = row.isSelected;
+    }
+  });
+};
+
+export default class MockAppMainGenericAdvancedTable extends Component<MockAppMainGenericAdvancedTableSignature> {
+  demoColumns = SAMPLE_COLUMNS;
+  @deepTracked demoModel: HdsAdvancedTableSignature['Args']['model'] = [
+    ...SAMPLE_MODEL,
+  ];
+
+  @action onSelectionChange({
+    selectionKey,
+    selectionCheckboxElement,
+    selectableRowsStates,
+  }: HdsAdvancedTableOnSelectionChangeSignature) {
+    console.log(...arguments);
+
+    if (selectionKey === 'all' && this.demoModel) {
+      const state = selectionCheckboxElement
+        ? selectionCheckboxElement.checked
+        : false;
+
+      updateModelWithSelectAllState(this.demoModel, state);
+    } else {
+      updateModelWithSelectableRowsStates(this.demoModel, selectableRowsStates);
+    }
+  }
+
+  <template>
+    <div class="mock-app-main-generic-advanced-table">
+      <HdsAdvancedTable
+        @columns={{this.demoColumns}}
+        @model={{this.demoModel}}
+        @hasStickyHeader={{true}}
+        @isSelectable={{true}}
+        @isStriped={{true}}
+        @onSelectionChange={{this.onSelectionChange}}
+        @hasStickyFirstColumn={{true}}
+      >
+        <:body as |B|>
+          {{! @glint-expect-error }}
+          <B.Tr @selectionKey={{get B.data "name"}}>
+            <B.Th>
+              <HdsLinkInline @href="www.google.com">
+                {{! @glint-expect-error }}
+                {{get B.data "name"}}
```

Comment:
Note: there's a lot of glint expect errors... B.data returns `Record<string, unknown>` which is causing issues.
* unknown isn't a valid type for glimmer component content 
* unknown doesn't match the type for any of the args that data from the model is being passed to (ex. `@selectionKey` on line 520)

Is there a better way we could type data? Or should consumers be typechecking the data that is returned from the Table/AdvancedTable components?